terraform/main.tf

```
@@ -0,0 +1,34 @@
+module "dns_zone" {
+  source = "./modules/dns"
+
+  domain = "domain.tld"
+  record_sets = [
+    {
+      name = "mysubdomain"
+      type = "A"
+      records = [
+        "1.2.3.4",
+        "5.6.7.8"
+      ]
+    },
+    {
+      name = "mysubdomain"
+      type = "AAAA"
+      records = [
+        "fe80::0001",
+        "fe80::0002"
+      ]
+    }
+  ]
+}
+
+module "record_sets" {
+  source = "./modules/dns//modules/record_sets"
+
+  domain = module.dns_zone.domain
```

```suggestion
  zone_domain = module.dns_zone.domain
```
Même remarque

De plus, pour pouvoir utiliser module.dns_zone.domain, il faut que celui-ci définisse un ouput:
```terraform
#modules/dns/main.tf
output "domain" {
 value =  dns_zone.this.domain
}
```

Cela permet en plus de s'assurer que la zone DNS est bien créée par le premier module avant d'utiliser le second, qui part du principe que la zone existe déjà (block data)